orchestrator/services/processes.py

```diff
@@ -526,6 +535,50 @@ def resume_process(
     return resume_func(process, user_inputs=user_inputs, user=user, broadcast_func=broadcast_func)
 
 
+def continue_awaiting_process(
+    process: ProcessTable,
+    *,
+    token: str,
+    input_data: State,
+) -> UUID:
+    """Continue a process awaiting data from a callback.
+
+    Args:
+        process: Process from database
+        token: The token which was generated for the process. This must match.
+        input_data: Data posted to the callback
+
+    Returns:
+        process id
+
+    """
+
+    pstat = load_process(process)
+    state = pstat.state.unwrap()
+
+    # Check if the token matches
+    token_from_state = state.get("__callback_token")
+    if token != token_from_state:
+        raise AssertionError("Invalid token")
+
+    # We need to pass the callback data to the worker executor. Currently, this is not supported.
+    # Therefore, we update the step state in the db and kick-off resume_workflow
+    # TODO: Allow passing additional data to be merged to the state upon resume_workflow
```

Comment:
Remove todo, looks done.
```suggestion
```

Reply:
It's not really done. The current way is an ugly workaround. We should pass the callback data to the resume_workflow task like it is done with user_inputs. Currently, resume_workflow only supports user_inputs and no additional data.

Reply:
Then fix it here or create ticket and add the ticket number to the todo. so we don't keep random todo's in the project